app/components/input-field.js

```diff
@@ -0,0 +1,31 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class InputFieldComponent extends Component {
+  @action handleInputFieldChange(event) {
+    let inputValue;
+    const { type, onChange, validator } = this.args;
+    const { value } = event.target;
+
+    switch (type) {
+      case 'string':
+        inputValue = value;
+        break;
+      case 'number':
+        inputValue = Number(value);
+        break;
+      case 'boolean':
+        inputValue = Boolean(value);
+        break;
+      default:
+        inputValue = value;
+        console.warn('Unknown type detected! Keeping the value as string');
+    }
+
+    if (typeof validator === 'function') {
+      validator(inputValue);
+    }
```

Comment:
I don't think we should pass the validator function into this component, only to be called on input value changed.

We are already calling the "onChange", that provides the value to the calling parent component. Let the parent component execute their validator to decide what they want to be doing with the new value.

Reply:
Makes sense